File: Task 3/C++Stock Viewer/Product/Furniture.cpp
#include "Furniture.h"

Furniture::Furniture() : Product(){
    material = "";
}

Furniture::Furniture(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string material)
    : Product(name, cost, supplier, discount_amount, packing_units, cost_per_unit){
    this->material = material; 
}

Furniture::Furniture(const Furniture& other) : Product(other) {
    this->material = other.material;
}

std::string Furniture::display_product_info(){
    return Product::display_product_info()+
           "\nMaterial: " + material;
}

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Furniture.java
package com.logistics.Components.Product;

public class Furniture extends Product {

    protected String material;

    public Furniture() {
        super();
        material = "";
    }

    public Furniture(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String material) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit);
        this.material = material;
    }

    // Copy Constructor
    public Furniture(Furniture other) {
        super(other);
        this.material = other.material;
    }

    // Clone method
    @Override
    public Furniture clone() {
        return new Furniture(this);
    }

    // Getters
    public String getMaterial() { return material; }

    // Setters
    public void setMaterial(String material) { this.material = material; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nMaterial: " + material;
    }
}
